### ISSUE TITLE
xUnit1039: Passing array into params errors

### ISSUE THREAD

**Issue by cheng93** [xunit/xunit#2827]
I believe passing an array to params input should be valid.

Looks to be caused by https://github.com/xunit/xunit.analyzers/pull/166/files

Works in xunit 2.6.1, but not 2.6.2

```csharp

public static TheoryData<int, string[]> PuzzleOneTestData
        = new TheoryData<int, string[]>()
        {
                {
                    3,
                    new []
                    {
                        "cpy 2 a",
                        "tgl a",
                        "tgl a",
                        "tgl a",
                        "cpy 1 a",
                        "dec a",
                        "dec a",
                    }
                }
        };

    [Theory]
    [MemberData(nameof(PuzzleOneTestData))]
    public void PuzzleOne(int expected, params string[] input)
    {
        var actual = Day23Solver.PuzzleOne(input);

        Assert.Equal(expected, actual);
    }
```

Here is branch instead if you want

https://github.com/cheng93/AdventOfCode/commit/28faf8c8f230d39a5df3c7624f2b5c6065597fef

`dotnet build` and you see the errors

**Comment by cheng93** [comment]
Actually in this scenario, you could argue why not remove params, which would fix it. However you could have the following scenario

```csharp
[Theory]
[MemberData(nameof(PuzzleOneTestData))]
[InlineData(1, "foo", "bar")]
public void PuzzleOne(int expected, params string[] input)
{
    var actual = Day23Solver.PuzzleOne(input);
    Assert.Equal(expected, actual);
}
```

**Comment by bradwilson** [comment]
Available in `1.7.0-pre.13`.
